src/reconcile/rec_visibility.c

```diff
@@ -245,26 +149,12 @@ int
 __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, void *ripcip,
   WT_CELL_UNPACK *vpack, WT_UPDATE_SELECT *upd_select)
 {
-    WT_CACHE *cache;
-    WT_CURSOR *las_cursor;
-    WT_CURSOR_BTREE *cbt;
-    WT_DECL_ITEM(key);
-    WT_DECL_RET;
-    WT_ITEM las_key, las_value;
-    WT_MODIFY_VECTOR modifies;
     WT_PAGE *page;
-    WT_TIME_PAIR las_start, las_stop;
-    WT_TXN_ISOLATION saved_isolation;
-    WT_UPDATE *birthmark_upd, *first_txn_upd, *first_inmem_upd, *inmem_upd_pos;
-    WT_UPDATE *tmp_upd, *upd;
-    wt_timestamp_t durable_ts, max_ts, mod_upd_ts;
-    size_t notused, upd_memsize;
+    WT_UPDATE *first_stable_upd, *first_txn_upd, *first_upd, *upd;
+    wt_timestamp_t max_ts;
+    size_t upd_memsize;
     uint64_t max_txn, txnid;
-    uint32_t las_btree_id, session_flags;
-    uint8_t *p, prepare_state, upd_type;
-    int cmp;
-    bool all_stable, las_cursor_open, las_positioned, list_prepared, list_uncommitted;
-    bool orig_val_appended, skipped_birthmark, sweep_locked, walked_past_sel_upd;
+    bool all_stable, list_prepared, list_uncommitted;
```

Comment:
I prefer to use the term committed, instead of stable, as we are operating on committed updates irrespective of whether they are later / earlier to stable timestamp.

Reply:
I get your point. We should adjust our view about the dirtiness of the page as well.

Reply:
We can't change it now since we haven't implemented inserting older versions to history store for update restore. We think it is better to do it later.